modules/dxp/apps/portal-search-similar-results/portal-search-similar-results-web/src/main/java/com/liferay/portal/search/similar/results/web/internal/portlet/shared/search/SimilarResultsPortletSharedSearchContributor.java

```diff
@@ -168,16 +173,12 @@ private void _filterByEntryClassName(
 			return;
 		}
 
-		Optional<String> classNameOptional = criteria.getTypeOptional();
+		String className = criteria.getType();
 
-		classNameOptional.ifPresent(
-			className -> {
-				if (!Validator.isBlank(className)) {
-					searchRequestBuilder.addComplexQueryPart(
-						_getComplexQueryPart(
-							_getEntryClassNameQuery(className)));
-				}
-			});
+		if ((className != null) && !Validator.isBlank(className)) {
```

Comment:
@slnn `Validator.isBlank` also checks whether the parameter is `null`, and returns true if `null`.